articles/quickstart/spa/vuejs/_includes/_centralized_login.md

```diff
@@ -2,475 +2,441 @@
+In this section, you will provide a way for the user to log in or log out, depending on their authentication status.
 
-<%= include('../../_includes/_callback_component') %>
-
-When a user authenticates at the login page and is then redirected back to your application, their authentication information will be contained in a URL hash fragment. The `handleAuthentication` method in the `AuthService` is responsible for processing the hash.
-
-Install `vue-router` to allow callbacks to be routed properly to the `Callback` component:
-
-```bash
-npm install vue-router
-```
-
-Create a component named `Callback` and populate it with a loading indicator. The component should also call `handleAuthentication` from the `AuthService`.
-
-```js
-<!-- src/components/Callback.vue -->
+Open the `views/Home.vue` file and modify the template to include two buttons that enable the user to log in and log out:
 
+```html
 <template>
-  <div>
-    <p>Loading...</p>
+  <div class="home">
+    <img alt="Vue logo" src="../assets/logo.png" />
+    <HelloWorld msg="Welcome to Your Vue.js App" />
+
+    <!-- Check that the SDK client is not currently loading before accessing is methods -->
+    <div v-if="!$auth.loading">
+      <!-- show login when not authenticated -->
+      <button v-if="!$auth.isAuthenticated" @click="login">Log in</button>
+      <!-- show logout when authenticated -->
+      <button v-if="$auth.isAuthenticated" @click="logout">Log out</button>
+    </div>
   </div>
 </template>
+```
 
+Notice that the buttons are wrapped in a directive that makes sure `$auth.loading` is false. This is because most methods on the SDK client are asynchronous, and you must wait until the client has finished loading before trying to access things like the user's authentication state.
+
+Further down the `Home.vue` file, modify the components script and implement the `login` and `logout` methods as follows:
+
+```js
 <script>
+// .. imports removed for brevity
+
 export default {
+  name: "home",
+  components: {
+    HelloWorld
+  },
   methods: {
-    handleLoginEvent(data) {
-      this.$router.push(data.state.target || "/");
+    // Log the user in
+    login() {
+      this.$auth.loginWithRedirect();
+    },
+    // Log the user out
+    logout() {
+      this.$auth.logout({
+        returnTo: window.location.origin
+      });
     }
-  },
-  created() {
-    this.$auth.handleAuthentication();
   }
 };
 </script>
 ```
 
-Add a new file `router.js` inside the `src` folder with the following content:
-
-```js
-// src/router.js
-
-import Vue from 'vue';
-import Router from 'vue-router';
-import Callback from './components/Callback';
+:::panel Checkpoint
+Try running the application at this point. Provided you have configured your Auth0 credentials correctly, you should be redirected to the Auth0 Universal Login page, and back to your app against once you have logged in.
```

Comment:
The callback mentioned at the top of the page - `http://localhost:3000/callback` - apparently is not correct. The one used is `http://localhost:3000` so I got a callback error when I ran it.

Reply:
Good catch, yes the `/callback` should now be removed.